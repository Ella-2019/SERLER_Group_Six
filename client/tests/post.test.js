/*import auth from './../auth/auth-helper.js'
import Post from './../post/Post.js'
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { configure, shallow, mount} from 'enzyme'
import chai, { expect } from 'chai'
import chaiEnzyme from 'chai-enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })
global.expect = expect
global.mount = mount
global.render = render
global.shallow = shallow*/

//import auth from './../auth/auth-helper.js'
import Post from '../post/Post.js'
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { configure, shallow, mount, render} from 'enzyme';
import chai, { expect } from 'chai';
import App from '../App';
import chaiEnzyme from 'chai-enzyme';

//jest.mock('./../auth/auth-helper.js') 
const dummyPostObject = {"_id":"5a3cb2399bcc621874d7e42f",
                         "postedBy":{"_id":"5a3cb1779bcc621874d7e428",
                         "name":"Joe"}, "text":"hey!",
                         "created":"2017-12-22T07:20:25.611Z",
                         "comments":[], "likes":[]} 
/*const dummyAuthObject = {user: {"_id":"5a3cb1779bcc621874d7e428",
                                "name":"Joe",
                                "email":"abc@def.com"}} */
                                
describe('App Component testing', function() {
    it('delete option visible only to authorized user', () => {
      const post = dummyPostObject 
      //const auth = dummyAuthObject 
      //jest.mock('./../auth/auth-helper.js') 
      //auth.isAuthenticated.mockReturnValue(auth) 
      const component = renderer.create(
        <MemoryRouter>
            <Post post={post} key={post._id} ></Post>
        </MemoryRouter>
      ) 

      let tree = component.toJSON() 
      expect(tree).toMatchSnapshot() 
    }) 

    chai.use(chaiEnzyme())

})