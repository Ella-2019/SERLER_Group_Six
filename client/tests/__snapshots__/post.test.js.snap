// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete option visible only to authorized user 1`] = `
<div
  className="MuiPaper-root-9 MuiPaper-elevation2-13 MuiPaper-rounded-10 MuiCard-root-8 Post-card-1"
>
  <div
    className="MuiCardHeader-root-36 Post-cardHeader-3"
  >
    <div
      className="MuiCardHeader-avatar-37"
    >
      <div
        className="MuiAvatar-root-42"
      >
        <img
          className="MuiAvatar-img-44"
          src="/api/users/photo/5a3cb1779bcc621874d7e428"
        />
      </div>
    </div>
    <div
      className="MuiCardHeader-content-39"
    >
      <span
        className="MuiTypography-root-45 MuiTypography-body2-53 MuiCardHeader-title-40"
      >
        <a
          href="/user/5a3cb1779bcc621874d7e428"
          onClick={[Function]}
        >
          Joe
        </a>
      </span>
      <span
        className="MuiTypography-root-45 MuiTypography-body2-53 MuiTypography-colorTextSecondary-67 MuiCardHeader-subheader-41"
      >
        Fri Dec 22 2017
      </span>
    </div>
    <div
      className="MuiCardHeader-action-38"
    >
      <button
        className="MuiButtonBase-root-75 MuiIconButton-root-69"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-74"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-77"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-83"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root-90 Post-cardContent-2"
  >
    <p
      className="MuiTypography-root-45 MuiTypography-body1-54 Post-text-4"
    >
      hey!
    </p>
  </div>
  <div
    className="MuiCardActions-root-91"
  >
    <button
      aria-label="Unlike"
      className="MuiButtonBase-root-75 MuiIconButton-root-69 MuiIconButton-colorSecondary-72 Post-button-7 MuiCardActions-action-92"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-74"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-77"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-83"
      />
    </button>
    <span
      className="MuiCardActions-action-92"
    >
      0
    </span>
    <button
      aria-label="Comment"
      className="MuiButtonBase-root-75 MuiIconButton-root-69 MuiIconButton-colorSecondary-72 Post-button-7 MuiCardActions-action-92"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-74"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-77"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-83"
      />
    </button>
    <span
      className="MuiCardActions-action-92"
    >
      0
    </span>
  </div>
  <hr
    className="MuiDivider-root-93"
  />
  <div>
    <div
      className="MuiCardHeader-root-36 Comments-cardHeader-97"
    >
      <div
        className="MuiCardHeader-avatar-37"
      >
        <div
          className="MuiAvatar-root-42 Comments-smallAvatar-98"
        >
          <img
            className="MuiAvatar-img-44"
            src="/api/users/photo/5a3cb1779bcc621874d7e428"
          />
        </div>
      </div>
      <div
        className="MuiCardHeader-content-39"
      >
        <span
          className="MuiTypography-root-45 MuiTypography-body2-53 MuiCardHeader-title-40"
        >
          <div
            className="MuiFormControl-root-103 MuiFormControl-marginNormal-104 Comments-commentField-99"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
          >
            <div
              className="MuiInput-root-107 MuiInput-formControl-108 MuiInput-multiline-113 MuiInput-underline-111"
            >
              <div
                className="Textarea-root-120"
                style={
                  Object {
                    "height": 19,
                  }
                }
              >
                <textarea
                  aria-hidden="true"
                  className="Textarea-shadow-122 Textarea-textarea-121"
                  readOnly={true}
                  rows="1"
                  tabIndex={-1}
                  value=""
                />
                <textarea
                  aria-hidden="true"
                  className="Textarea-shadow-122 Textarea-textarea-121"
                  readOnly={true}
                  rows={1}
                  tabIndex={-1}
                  value=""
                />
                <textarea
                  aria-invalid={false}
                  aria-required={false}
                  className="Textarea-textarea-121 MuiInput-input-115 MuiInput-inputMultiline-117"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Write something ..."
                  rows={1}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;
